### PR TITLE
[android] Fix copied to clipboard not showing when on lock screen

### LINKED ISSUE #6938: [android] Copied to clipboard toast is not displayed when running on the lock screen
1. Run OM
2. Make sure "Use on the lock screen" OM setting is enabled.
3. Lock the device
4. Unlock the device
5. Select any object on the map
6. Try to hold your finger on the title of the selected object
7. Observe that "Copied to clipboard" message is not displayed

Reproduced on Android 14, didn't test on other versions.

### PR DESCRIPTION
**Pull request by arjundevasi**
The issue stated that the app does not show any visual feedback when copied place name, description or wikipedia from lock screen.

Made code changes at three places which will fix the issue.

Fixes #6938

### CODE REVIEW

**Comment by biodranik** [comment]
Is the text "Copied" now displayed on the lock screen after your fix?

**Comment by arjundevasi** [comment]
Yes, a snack bar with the text - copied to clipboard: "string" is shown.
Screenshot_20240126-071342

File: android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java [hunk 1]
```diff
@@ -79,7 +80,11 @@ static String toString(@BottomSheetBehavior.State int state)
   public static void copyToClipboard(Context context, View frame, String text)
   {
     Utils.copyTextToClipboard(context, text);
-    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU)
+
+    KeyguardManager keyguardManager = (KeyguardManager) context.getSystemService(Context.KEYGUARD_SERVICE);
+
+    //only show snackBar if device is locked or android sdk is earlier than TIRAMISU.
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU || Build.VERSION.SDK_INT > Build.VERSION_CODES.TIRAMISU && keyguardManager.isDeviceLocked())
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java:87]: isDeviceLocked available starting from LOLLIPOP_MR1 (API 22), while Organic Maps works from LOLLIPOP (API 21). Why is TIRAMISU (API 33) check needed here?

Comment (vng) [inline android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java:87]: Please, put braces, the conditions order is not obvious here:
```if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU || Build.VERSION.SDK_INT > Build.VERSION_CODES.TIRAMISU && keyguardManager.isDeviceLocked())```

- What's wrong with ```Build.VERSION.SDK_INT == Build.VERSION_CODES.TIRAMISU```? It doesn't support a snack bar at all?
- ```Build.VERSION.SDK_INT > Build.VERSION_CODES.TIRAMISU``` supports a snack bar in locked mode only?

Comment (arjundevasi) [reply android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java:87]: apologies, working on the fix.

Comment (arjundevasi) [reply android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java:87]: the snack bar should be shown only on lockscreen for android version 13 and higher, when unlocked the preview popup is shown by default.

for android version lower than 13, the snackbar should be shown regardless of device is locked or not.

The if condition needs fixing, working on it.

Comment (arjundevasi) [reply android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java:87]: how about this modification: 
``` if(Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU)
      {
        Utils.showSnackbarAbove(view.getRootView().findViewById(R.id.pp_buttons_layout), view,
                                ctx.getString(R.string.copied_to_clipboard, phoneNumber));
      }
      else if (keyguardManager.isDeviceLocked())
      {
        Utils.showSnackbarAbove(view.getRootView().findViewById(R.id.pp_buttons_layout), view,
                                ctx.getString(R.string.copied_to_clipboard, phoneNumber));
      } ```

Comment (arjundevasi) [reply android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java:87]: how about this modification: 
``` 
if(Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU)
      {
        Utils.showSnackbarAbove(view.getRootView().findViewById(R.id.pp_buttons_layout), view,
                                ctx.getString(R.string.copied_to_clipboard, phoneNumber));
      }
      else if (keyguardManager.isDeviceLocked())
      {
        Utils.showSnackbarAbove(view.getRootView().findViewById(R.id.pp_buttons_layout), view,
                                ctx.getString(R.string.copied_to_clipboard, phoneNumber));
      }
 ```

Comment (Jean-BaptisteC) [reply android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java]:
```
if(Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU || keyguardManager.isDeviceLocked())
      {
        Utils.showSnackbarAbove(view.getRootView().findViewById(R.id.pp_buttons_layout), view,
                                ctx.getString(R.string.copied_to_clipboard, phoneNumber));
      }
```

Comment (arjundevasi) [reply android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java]: this is so much concise, you should create this pull request.

File: android/app/src/main/java/app/organicmaps/widget/placepage/sections/PlacePhoneAdapter.java [hunk 2]
```diff
@@ -20,9 +21,12 @@ public class PlacePhoneAdapter extends RecyclerView.Adapter<PlacePhoneAdapter.Vi
 
   private ArrayList<String> mPhoneData = new ArrayList<>();
 
-  public PlacePhoneAdapter() {}
+  public PlacePhoneAdapter()
+  {
+  }
```
[the remark this answers is not in the document]
Comment (arjundevasi) [reply android/app/src/main/java/app/organicmaps/widget/placepage/sections/PlacePhoneAdapter.java:26]: understood, working on it.

[the remark this answers is not in the document]
Comment (arjundevasi) [reply android/app/src/main/java/app/organicmaps/widget/placepage/sections/PlacePhoneAdapter.java]: The change on this part was done by IDE when I clicked on  Reformat Code

[the remark this answers is not in the document]
Comment (biodranik) [reply android/app/src/main/java/app/organicmaps/widget/placepage/sections/PlacePhoneAdapter.java]: We do not merge any random ide changes. Please learn how to amend and rebase commits to fix them. Or add newer commits on top so we can squash it into one commit later.

**Comment by arjundevasi** [comment]
I have made changes as requested and committed them. I thank you all for guiding me and having patience with me. 
Please review them and if they are not correct please let me know so i can close this pull request and make a fresh one without any clutters. Thank you.

File: android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java [hunk 3]
```diff
@@ -79,7 +80,10 @@ static String toString(@BottomSheetBehavior.State int state)
   public static void copyToClipboard(Context context, View frame, String text)
   {
     Utils.copyTextToClipboard(context, text);
-    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU)
+
+    KeyguardManager keyguardManager = (KeyguardManager) context.getSystemService(Context.KEYGUARD_SERVICE);
+    //only show snackBar if device is locked or android sdk is earlier than TIRAMISU.
```
[the remark this answers is not in the document]
Comment (biodranik) [reply android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java:85]:
```suggestion
    // Starting from API 33, the automatic system control that shows copied text is displayed.
```

File: android/app/src/main/java/app/organicmaps/widget/placepage/sections/PlacePhoneAdapter.java [hunk 4]
```diff
@@ -88,7 +88,10 @@ public boolean onLongClick(View view)
       final String phoneNumber = mPhone.getText().toString();
       final Context ctx = view.getContext();
       Utils.copyTextToClipboard(ctx, phoneNumber);
-      if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU)
+
+      KeyguardManager keyguardManager = (KeyguardManager) ctx.getSystemService(Context.KEYGUARD_SERVICE);
+      //only show snackBar if device is locked or android sdk is earlier than TIRAMISU.
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/widget/placepage/sections/PlacePhoneAdapter.java:93]: ditto

File: android/app/src/main/java/app/organicmaps/widget/placepage/sections/PlacePageBookmarkFragment.java [hunk 5]
```diff
@@ -137,7 +138,10 @@ public boolean onLongClick(View v)
 
     final Context ctx = requireContext();
     Utils.copyTextToClipboard(ctx, notes);
-    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU)
+
+    KeyguardManager keyguardManager = (KeyguardManager) ctx.getSystemService(Context.KEYGUARD_SERVICE);
+    //only show snackBar if device is locked or android sdk is earlier than TIRAMISU.
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/widget/placepage/sections/PlacePageBookmarkFragment.java:143]: ditto

**Review by biodranik — APPROVED** [review]
Thank you for the contribution!

**Pull request merged**